lib/devices/ios/ios-log.js

```diff
@@ -44,91 +40,112 @@ IosLog.prototype.startCapture = function (cb) {
+          if (_.isUndefined(this.simUdid)) {
+            return cb(new Error("iOS8 log capture requires a sim udid"));
           }
-        ], function (err) { cb(err); });
-      };
-      async.waterfall([
-        function (cb) {
-          if (logsPath.indexOf('*') < 0) {
-            // there is no doubt on system.log location
-            touchLog(cb);
-          } else cb();
-        }, function (cb) {
-          glob(path.resolve(logsPath, "system.log"), function (err, files) {
-            if (err || files.length < 1) {
-              logger.error("Could not start log capture because no iOS " +
-                "simulator logs could be found at " + logsPath + "/system.log" +
-                "Logging will not be functional for this run");
-              return cb(new Error('Could not start log capture'));
-            }
-            cb(null, files);
-          });
-        }, function (files, cb) {
-          var lastModifiedLogPath = files[0]
-            , lastModifiedLogTime = fs.statSync(files[0]).mtime;
-          _.each(files, function (file) {
-            var mtime = fs.statSync(file).mtime;
-            if (mtime > lastModifiedLogTime) {
-              lastModifiedLogPath = file;
-              lastModifiedLogTime = mtime;
+          logsPath = path.resolve(process.env.HOME, "Library", "Logs",
+                                  "CoreSimulator", this.simUdid);
+        } else {
+          logger.debug("Starting iOS 7.* simulator log capture");
+          logsPath = path.resolve(process.env.HOME, "Library", "Logs",
+                                  "iOS Simulator", "7.*");
+        }
+        var errOnlyCb = function (cb) { return function (err) { cb(err); }; }; // makes waterfall safer
+        var touchLog = function (cb) {
+          async.series([
+            function (cb) { mkdirp(logsPath, errOnlyCb(cb)); },
+            function (cb) { touch(path.resolve(logsPath, "system.log"), errOnlyCb(cb)); },
+            function (cb) {
+              require('fs').appendFile(
+                path.resolve(logsPath, "system.log"),
+                'A new simulator is about to start!',
+                errOnlyCb(cb));
             }
-          });
-          this.proc = spawn("tail", ["-f", "-n", "1", lastModifiedLogPath]);
-          // -n 1 is used so that tail returns a line that lets this process know that the
-          // first line returned wasn't from stderr so it can tell the tailing was successful
-          this.finishStartingLogCapture(cb);
-        }.bind(this)
-      ], function () {
-        logger.warn('System log capture failed.');
-        // if system log fails this is no big deal
-        cb();
-      });
-    } else {
-      logger.debug("Starting iOS 6.* simulator log capture");
-      this.proc = spawn("tail", ["-f", "-n", "1", "/var/log/system.log"]);
-      this.finishStartingLogCapture(cb);
-    }
+          ], function (err) { cb(err); });
+        };
+        async.waterfall([
```

Comment:
I'm really not a fan of big `async.series` or `async.waterfall` blocks that define the functions inline. It is really messy to read. I know you didn't initiate this pattern here in this file, but since we're refactoring, can we clean it up? I'd prefer to define each function ahead of time, e.g.:

```js
var prepLogPath = function (cb) {...};
var findSimLogs = function (cb) {...};
var tailSimLogs = function (files, cb) {...};
async.waterfall([prepLogPath, findSimLogs, tailSimLogs], function (err) {
  if (err) {
    logger.warn('System log capture failed');
  }
  cb();
});
```

(also note how I think I fixed a bug by only outputting the log warning when there was an error)